lib/shrinkwrap.js

```diff
@@ -31,10 +34,37 @@ function shrinkwrap (args, silent, cb) {
 
   var dir = path.resolve(npm.dir, '..')
   npm.config.set('production', true)
-  readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
-    var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
-    shrinkwrap_(pkginfo, silent, cb)
-  })))
+
+  readPackage(iferr(cb, function (data) {
+    lifecycle(data, 'preshrinkwrap', function () {
+      readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
+        var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
+
+        chain([
+          [lifecycle, tree.package, 'shrinkwrap'],
+          [shrinkwrap_, pkginfo, silent],
+          [lifecycle, tree.package, 'postshrinkwrap']
+        ], iferr(cb, function (data) {
+          cb(null, data[0])
+        }))
+      })))
+    })
+  }))
+}
+
+function readPackage (cb) {
```

Comment:
Please use `read-package-json` instead of this. (And yeah, I see that `lib/version.js` _isn't_– we must have missed that during code review. =/) Using it would look like:
```
var readPackageJson = reaquire('read-package-json')

var packagePath = path.join(npm.localPrefix, 'package.json')
readPackageJson(packagePath, iferr(cb, function (data) {
…
}))
```